internal/metrics/otel_test.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+package metrics
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/nginx/agent/v3/internal/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGRPCExporter_Constructor(t *testing.T) {
+	conf := config.Metrics{
+		OTelExporterTarget: "dummy-target",
+		ReportInterval:     1 * time.Second,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	_, err := NewGRPCExporter(ctx, conf)
```

Comment:
can you assert on anything in otlpmetricgrpc.Exporter return?

Reply:
The `Exporter` struct only has private fields, so can't really assert against them.
```go
// Exporter is a OpenTelemetry metric Exporter using gRPC.
type Exporter struct {
	// Ensure synchronous access to the client across all functionality.
	clientMu sync.Mutex
	client   interface {
		UploadMetrics(context.Context, *metricpb.ResourceMetrics) error
		Shutdown(context.Context) error
	}

	temporalitySelector metric.TemporalitySelector
	aggregationSelector metric.AggregationSelector

	shutdownOnce sync.Once
}
```
I suppose I can add a nil check, since it returns a pointer to the `Exporter`.